LEAF_Request_Portal/admin/templates/mod_workflow.tpl

```diff
@@ -1210,62 +1211,104 @@
+
+            $('#fillDependency').val(res[0].fillDependency);
+            document.getElementById('backwards_action_note').style.display = parseInt(res[0].fillDependency) < 0 ? 'block': 'none';
+            document.getElementById('fillDependency').addEventListener('change', actionDirectionNote);
+
+            dialog.setSaveHandler(function() {
+                let sort = parseInt($('#actionSortNumber').val());
+                sort = Number.isInteger(sort) ? sort : 0;
+                sort = sort < -128 ? -128
+                        : sort > 127 ? 127
+                        : sort;
+                $.ajax({
+                    type: 'POST',
+                    url: '../api/workflow/editAction/_' + actionType,
+                    data: {
+                        actionText: $('#actionText').val(),
+                        actionTextPasttense: $('#actionTextPasttense').val(),
+                        actionIcon: $('#actionIcon').val(),
+                        sort: sort,
+                        fillDependency: $('#fillDependency').val(),
+                        CSRFToken: CSRFToken
+                    },
+                    success: function() {
+                        listActionType();
+                    },
+                    error: (err) => console.log(err),
+                });
+                dialog.hide();
+            });
+        });
+    }
+
+    function getAction(actionType, callback) {
         $.ajax({
             type: 'GET',
             url: '../api/workflow/action/_' + actionType,
             success: function(res) {
-                dialog.indicateIdle();
-                dialog.setContent(renderActionInputModal(res[0]));
-                $('#fillDependency').val(res[0].fillDependency);
-                document.getElementById('backwards_action_note').style.display = parseInt(res[0].fillDependency) < 0 ? 'block': 'none';
-                document.getElementById('fillDependency').addEventListener('change', actionDirectionNote);
-                dialog.setSaveHandler(function() {
-                    let sort = parseInt($('#actionSortNumber').val());
-                    sort = Number.isInteger(sort) ? sort : 0;
-                    sort = sort < -128 ? -128
-                         : sort > 127 ? 127
-                         : sort;
+                callback(res);
+            },
+            error: (err) => console.log(err),
+            cache: false
+        });
+    }
+
+    //deletes action type
+    function deleteActionType(actionType) {
+        // find out if this action is being used in a workflow currently
+        getUsedActionType(actionType, function (res) {
+            console.log(res);
+            let workflows = '';
+
+            for (let i in res.data) {
+                workflows += res.data[i].description + "<br />";
+            }
+
+            if (res.status.code == 2 && res.data.length) {
+                console.log('something to display');
```

Comment:
Did you want to leave this in?  If so, perhaps something more descriptive?

Reply:
Cleaned up.